### ISSUE TITLE
bug: sync k8s crd resources to apisix at intervals, causing etcd dbsize to skyrocket

### ISSUE THREAD

**Issue by Donghui0** [apache/apisix-ingress-controller#1646]
### Issue description

The controller parameter `apisix-resource-sync-interval` is used to control whether the k8s crd resource is consistent with apisix(etcd). 
However, when the number of resources (route/upstream/pods)  is too large, the sync of the interval will have a high impact on the load of etcd service. And it will continue to increase etcd dbsize.

See below, the dbsize increase to 2GB within 7 days, causing etcd to be unable to write new data.   (The etcd cluster consists of 3 nodes,  4c16g)
<img width="1462" alt="image" src="https://user-images.githubusercontent.com/7134928/216584450-84371216-e9d4-46d3-8d23-34a72a6d0cfa.png">

By compressing the data regularly and reducing the dbsize, you can see that the dbsize of etcd will still increase at intervals. The interval is basically consistent with the parameter apisix-resource-sync-interval, which is 600s.
<img width="1401" alt="image" src="https://user-images.githubusercontent.com/7134928/216581895-dad78201-0010-4b1b-bb3e-6749ffc4155e.png">

### Environment

apisix-ingress-controller version: 1.5.0
number resource in apisix:     ApisixRoutes  300+,   Upstream 20+,  Upstream.node  60+.

### Minimal test code / Steps to reproduce

Under 300+ routing entries, when the number of Upstream Nodes is large, it can be reproduced by running at intervals.

params  apisix-resource-sync-interval  set to  300s or 600s.

### Actual result

etcd dbsize increase.

### Error log

none

**Comment by tao12345666333** [comment]
As far as I know @AlinsRan  is already following it up

**Comment by Donghui0** [comment]
modify  apisix-ingress-controller  code, remove `resourceSyncLoop` feature.  etcd dbsize becomes stable.
![image](https://user-images.githubusercontent.com/7134928/216880186-796199af-cf95-4928-832b-3c1a3264b66a.png)

etcd version: v3.5.0  (enable simple token auth)
apisix version: v2.14.1
ingress-controller version: v1.5.0

**Comment by machinly** [comment]
You can also enable auto compaction of etcd to mitigate memory increase.
Specifically, add follow env on etcd pod.
```
- name: ETCD_AUTO_COMPACTION_MODE
  value: "periodic"
- name: ETCD_AUTO_COMPACTION_RETENTION
  value: "1h"
```

**Comment by Donghui0** [comment]
> 

Yes, I can do that. But don't you think etcd's dbsize grows abnormally ?

**Comment by machinly** [comment]
I agree that this is abnormally, auto compaction is just a mitigating measure to avoid ETCD downtime.

**Comment by AlinsRan** [comment]
![image](https://user-images.githubusercontent.com/79972061/216918824-2c120cc3-6210-4584-ac77-81667953a1f0.png)

Even if the resource has not changed, data will be written to the disk every few minutes.

I think maybe we can add compare logic in APISIX or Controller to fix it.

**Comment by tao12345666333** [comment]
I will put this to the v1.7 milestone.

**Comment by tao12345666333** [comment]
#1685 A configuration item has been added to allow users to turn it off.

This feature itself is a compensating mechanism to avoid data in APISIX being altered by other means. If the user does not need this function, it can be turned off (set to 0)

**Comment by tao12345666333** [comment]
#1685  and https://github.com/apache/apisix-ingress-controller/pull/1742 has been merged. 

This issue has already been resolved and will be released in v1.7. I will close this issue now. If you have any further ideas, please feel free to reply.
